### PR TITLE
Clean up the sanitizer runs

### PR DESCRIPTION
**Pull request by jbachorik**
**What does this PR do?**:
This is trying to calm down the rather noisy nightly sanitized runs.

**Motivation**:
Prevent 'warning' fatigue

**Additional Notes**:
I had to disable TSAN completely for nightlies. After adding a bunch of suppressions for various JVM related libraries most of the tests started either failing assertions or never finishing.
TSAN can still be run locally via `./gradlew testTsan` if you are on a linux box and tsan is fully installed (I could not get it working on the workspace box, though - it's still missing some tsan related *.so file which I have no idea where to get) - most likely it will make the unit tests failed because of messed up timing, or plainly livelock them, but you can try.

**How to test the change?**:

**For Datadog employees**:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles
  credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.
- [x] JIRA: [PROF-10035]

Unsure? Have a question? Request a review!

[PROF-10035]: https://datadoghq.atlassian.net/browse/PROF-10035?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ddprof-lib/src/main/cpp/livenessTracker.cpp [hunk 1]
```diff
@@ -44,44 +44,47 @@ void LivenessTracker::cleanup_table(bool forced) {
     JNIEnv* env = VM::jni();
     jvmtiEnv* jvmti = VM::jvmti();
 
-    u64 start = OS::nanotime(), end;
+    int epoch_diff = (int) (target_gc_epoch - current);
 
     _table_lock.lock();
-    int epoch_diff = (int) (target_gc_epoch - current);
```
Comment (richardstartin) [inline ddprof-lib/src/main/cpp/livenessTracker.cpp:50]: is this change necessary?

Comment (jbachorik) [reply ddprof-lib/src/main/cpp/livenessTracker.cpp:50]: The whole block change was triggered by division by zero in the debug statement. While that is unlikely, getting the table of size 0 is still possible and moving the check for the non-zero check before doing anything more expensive seemed like an ok idea.

**Review by richardstartin — APPROVED** [review]
